Hotfix action (#8192)
cloudflare/workers-sdk: CONTRIBUTING.md

## Releases

We generally cut Wrangler releases on Tuesday & Thursday each week. If you need a release cut outside of the regular cadence, please reach out to the [@cloudflare/wrangler-admins](https://github.com/orgs/cloudflare/teams/wrangler-admins) team.

### Hotfix releases

> Only members of `@cloudflare/wrangler` can trigger a hotfix. In general, a hotfix release should be treated as a solution of last resort—before use, please first check whether the fix can be applied and released using the regular Version Packages release flow.

Reviewer: ```suggestion
Only members of `@cloudflare/wrangler` can trigger a hotfix release. In general, a hotfix release should be treated as a solution of last resort—before use, please first check whether the fix can be applied and released using the regular Version Packages release flow.
```

If a hotfix release of Wrangler, Miniflare, or C3 is required, you should:

- Prepare a hotfix release PR: checkout the previous release of `workers-sdk`, apply the changes that should be in the hotfix, and then increment the patch version of the packages that should be released as part of the hotfix

- Get approvals for that PR, and make sure CI checks are passing

> - Manually trigger a hotfix release from that PR using the "Release a hotfix" GitHub action. Optionally, you can first publish it to the `hotfix` tag on NPM in order to verify the release.

Reviewer: is it possible to link to where this action lives/or should be trigerred from?

Author: https://github.com/cloudflare/workers-sdk/pull/8192/commits/266068f5fd1458720c93082c8876415a3e8f26e0

> - _[CRUCIAL]_ Once the hotfix release is out and verified, merge the fixes into main before the next regular release of `workers-sdk`.

Reviewer: I don't suppose we could automate this somehow 🤔 ... as in if a hotfix PR is released, a new PR to `main` with hotfix changes would be automatically created. Totally not in the scope of this PR...just thinking out loud

Reviewer 2: ooh yes +1

Author: Yeah! Probably as a followup though—I'd love to get this merged and tested first

## Releases

We generally cut Wrangler releases on Tuesday & Thursday each week. If you need a release cut outside of the regular cadence, please reach out to the [@cloudflare/wrangler-admins](https://github.com/orgs/cloudflare/teams/wrangler-admins) team.

### Hotfix releases

> Only members of the `@cloudflare/wrangler` can trigger a hotfix. In general, a hotfix is the last solution that should be reached for—if possible, check to see if your fix can be applied and then released using the regular Version Packages release flow.

Reviewer 2: ```suggestion
Only members of `@cloudflare/wrangler` can trigger a hotfix. In general, a hotfix is the last solution that should be reached for—if possible, check to see if your fix can be applied and then released using the regular Version Packages release flow.
```

> Only members of the `@cloudflare/wrangler` can trigger a hotfix. In general, a hotfix is the last solution that should be reached for—if possible, check to see if your fix can be applied and then released using the regular Version Packages release flow.

Reviewer 2: ```suggestion
Only members of `@cloudflare/wrangler` can trigger a hotfix. In general, a hotfix release should be treated as a solution of last resort -- before use, please first check whether the fix can be applied and released using the regular Version Packages release flow.
```

If a hotfix release of Wrangler, Miniflare, or C3 is required, you should:

> - Prepare a PR which captures what should be in the hotfix
> 
>   - In general, the best way to do this is to checkout the previous release of `workers-sdk`, apply the changes you need, and then increment the patch version of the packages that should be released as part of this hotfix.

Reviewer 2: ```suggestion
- Prepare a hotfix release PR: checkout the previous release of `workers-sdk`, apply the changes that should be in the hotfix, and then increment the patch version of the packages that should be released as part of the hotfix
```

- Get approvals for that PR, and make sure CI checks are passing

> - Manually trigger a hotfix release from that PR using the "Release a hotfix" GitHub action. Optionally, you can first publish it to the `hotfix` tag on NPM in order to verify the release.

Reviewer 2: > "Optionally, you can first publish it to the `hotfix` tag on NPM in order to verify the release.
Comment"

is there a reason this shouldn't/can't be done? if no, i suggest we remove the "optionally" wording and explicitly direct to do this

Author: It's probably not really necessary, given we can test with the pre-release on the hotfix PR branch. We can likely remove this after we've merged and tested the hotfix workflow, but we need it for now so that we can run a test hotfix without overwriting npm latest

Reviewer 2: 👍

> - Once the release is out and verified, it's _crucial_ that the fixes are then merged into main before the next regular release of `workers-sdk`. Additionally, make sure that the version number of the next changesets-based release of Wrangler/Miniflare/C3 is greater than the version used for the hotfix. The easiest way to do this is to add a dummy `minor` changeset entry for each of the packages that had a hotfix published.

Reviewer 2: ```suggestion
- _[CRUCIAL]_ Once the hotfix release is out and verified, merge the fixes into main before the next regular release of `workers-sdk`. 
- Make sure that the version number of the next changesets-based release of Wrangler/Miniflare/C3 is greater than the version used for the hotfix: add a dummy `minor` changeset entry for each of the packages that had a hotfix published.
```
